CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# How to contribute
+
+## Getting Started
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue, assuming one does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+
+## Making patches
+
+* Read [Manual (How to write packages for Nix)](https://nixos.org/nixpkgs/manual/)
+* Fork the repository on GitHub.
+* Create a branch for your future fix.
+  * You can make branch from a commit of your local `nixos-version`. That will help you to avoid additional local compilations. Because you will recieve some packages from cache server.
+    * For example: `nixos-version` returns `15.05.git.0998212 (Dingo)`. So you can do:
+
+        ```bash
+        git checkout 0998212
+        git checkout -b 'fix/pkg-name-update'
+        ```
+    * Or use `$(git rev-parse --show-cdup)maintainers/scripts/update-channel-branches.sh` to pull latest versions of the channel as branches. Run it anywhere in your working directory of nixpkgs.
+  * Please avoid working directly on the `master` branch.
+* Make commits of logical units. 
```

Comment:
This one should be a bit more verbose. Git commits can be used as a timeline of work or as "working units" that serve as documentation (that's the one we want encourage).

I'd also mention that mostly new packages/updates are squashed into a single commit.